JSON/requirements-sample-fio.json

```diff
@@ -0,0 +1,38 @@
+{
+    "defaults": {
+        "rw": ["read", "randread"],
+        "bs": ["16k"]
+    },
+    "essentials": {
+        "write_iops_log": ["fio"],
+        "write_lat_log": ["fio"]
+    },
+    "presets": [
+        {
+            "name": "sequential-read",
+            "params": [
+                { "arg": "rw", "values": [ "read" ] },
+                { "arg": "bs", "values": [ "4k" ] }
+            ]
+        }
+    ],
```

Comment:
I'm wondering if there is a reason that we have the `params` formatted differently in `defaults` and `essentials` than we do in `presets`?

Reply:
IMHO `presets` should be just like the other 2. You can specify as many params you want inside presets. In other words, it does not need to be an array like we have in mv-params.json.
So back to my original question, should we care about "named" elements containing an array of params ? I don't see where you would use "sequential-read". The actual params are `rw` and `bs`.
Can I get rid of name/params keypair ?

Reply:
I think the idea behind `presets` is to be able to do something like this:

```
crucible run <benchmark> preset=<preset-name>
```

So we can't do this today, but this is the type of thing we are wanting to enable via `multiplex`.

As for the `params` formatting...all the existing logic in `multiplex.py` is centered around taking parameters in this format:

```
{ "arg": "foo", "values": [ "bar", "zorg" ] }
```

and multiplexing them with the other parameters in a defined parameter set.  If we don't maintain that format in JSON input files we are either going to have to expand the parameter formats that `multiplex.py` can handle OR we will have to internally convert the various external formats into the format that is currently understood.  It seems to me that the easiest and least confusing way is to just use the same format everywhere.

Reply:
```
  1 {        
  2     "presets": {
  3         "essentials": [
  4             { "arg": "write_iops_log", "values": ["fio"] },
  5             { "arg": "write_lat_log", "values": ["fio"] }
  6         ],
  7         "defaults": [
  8             { "arg": "rw", "values": ["read", "randread"] },
  9             { "arg": "bs", "values": ["16k"] }
 10         ],
 11         "sequential-read": [
 12             { "arg": "rw", "values": [ "read" ] },
 13             { "arg": "bs", "values": [ "4k" ] }
 14         ]
 15     },   
 16     "validations": {
 17         "size_KMG" : {
 18             "description" : "bytes in k/K (1024), m/M (1024^2) or g/G (1024^3): 4k 16M 1g",                                                                                                                    
 19             "arguments" : [ "bs" ],
 20             "value" : "[0-9]+[kbmgKBMG]",
 21             "transform" : [ "s/([0-9]+)[gG]/($1*1024).\"M\"/e",
 22                             "s/([0-9]+)[mM]/($1*1024).\"K\"/e" ]
 23         },   
 24         "log_types" : {
 25           "description" : "all possible log types",
 26           "arguments" : [ "write_bw_log", "write_hist_log", "write_iolog", "write_iops_log", "write_lat_log" ],
 27           "value" : "^fio$"
 28         },
 29         "rw_types" : {
 30           "description" : "all possible testtypes",
 31           "arguments" : [ "rw" ],
 32           "value" : "^(|rand)(read|write|trim)$|^readwrite$|^randrw$|^trimwrite$"
 33         }
 34     }    
 35 } 
```

Are you ok with this ?

Reply:
I think that looks pretty good.  What do you think @atheurer ?